worker-queue-rabbit/src/test/java/com/hpe/caf/worker/queue/rabbit/WorkerConfirmListenerTest.java

```diff
@@ -224,7 +256,94 @@ public void testDuplicateRegister()
     {
         BlockingQueue<Event<QueueConsumer>> q = new LinkedBlockingQueue<>();
         WorkerConfirmListener conf = new WorkerConfirmListener(q);
-        conf.registerResponseSequence(1, 100);
-        conf.registerResponseSequence(1, 100);
+        conf.registerResponseSequence(1, new RabbitTaskInformation("100"));
+        conf.registerResponseSequence(1, new RabbitTaskInformation("100"));
+    }
+    
+    @Test
+    public void testAckSingleTaskMultiplePublish()
+        throws IOException, InterruptedException
+    {
+        BlockingQueue<Event<QueueConsumer>> q = new LinkedBlockingQueue<>();
+        WorkerConfirmListener conf = new WorkerConfirmListener(q);
+        RabbitTaskInformation rabbitTaskInfo_100 = new RabbitTaskInformation("100");
+        rabbitTaskInfo_100.incrementResponseCount(false);
+        conf.registerResponseSequence(5, rabbitTaskInfo_100);
+        rabbitTaskInfo_100.incrementResponseCount(false);
+        conf.registerResponseSequence(1,rabbitTaskInfo_100);
+        rabbitTaskInfo_100.incrementResponseCount(true);
+        conf.registerResponseSequence(2, rabbitTaskInfo_100);
+        conf.handleAck(5, true);
+        Event<QueueConsumer> e = q.poll(1000, TimeUnit.MILLISECONDS);
+        Assert.assertNotNull(e);
+        Assert.assertTrue(e instanceof ConsumerAckEvent);
+        Assert.assertEquals(100L, ((ConsumerAckEvent) e).getTag());
+    }
+    
+    @Test
+    public void testNackSingleTaskMultiplePublish()
+        throws IOException, InterruptedException
+    {
+        BlockingQueue<Event<QueueConsumer>> q = new LinkedBlockingQueue<>();
+        WorkerConfirmListener conf = new WorkerConfirmListener(q);
+        RabbitTaskInformation rabbitTaskInfo_100 = new RabbitTaskInformation("100");
+        rabbitTaskInfo_100.incrementResponseCount(false);
+        conf.registerResponseSequence(5, rabbitTaskInfo_100);
+        rabbitTaskInfo_100.incrementResponseCount(false);
+        conf.registerResponseSequence(1,rabbitTaskInfo_100);
+        rabbitTaskInfo_100.incrementResponseCount(true);
+        conf.registerResponseSequence(2, rabbitTaskInfo_100);
+        conf.handleNack(4, true);
+        Event<QueueConsumer> e = q.poll(1000, TimeUnit.MILLISECONDS);
```

Comment:
Is it worth verifying null is returned from q.poll after the relevant calls to registerResponseSequence. Same comment applies to other tests below.

Reply:
yes Andy, in the next conf.handleAck(5, true); in line 303... you will not get any event as a negative ACK is already sent. Found it to be valid.